client/app/models/package.js

```diff
@@ -77,7 +77,8 @@ export default class PackageModel extends Model {
     if (this.dcpPackagetype === DCPPACKAGETYPE.RWCDS.code) {
       await this.rwcdsForm.save();
     }
-    if (this.dcpPackagetype === DCPPACKAGETYPE.DRAFT_LU_PACKAGE.code) {
+    if (this.dcpPackagetype === DCPPACKAGETYPE.DRAFT_LU_PACKAGE.code
+      || this.dcpPackagetype === DCPPACKAGETYPE.FILED_LU_PACKAGE.code) {
```

Comment:
Lots of complexity in this method. It's starting to look like it needs to a [polymorphic model](https://guides.emberjs.com/release/models/relationships/#toc_polymorphism).

Is there a way we could start making these computed properties with human-readable names that describe what the conditional is?

Reply:
ohh interesting idea. With polymorphism I'm guessing we can do away with these conditionals and instead parameterize which form is saved. 
```
// models/package.js
export default class PackageModel extends Model {
  @action
   save(){
       this.form.save();
   }
}
```

```
// models/draft-landuse-package.js
export default class DraftLandUsePackageModel extends Model {
  @belongsTo('landuse-form', { async: false })
  form 
}



```

Reply:
> Is there a way we could start making these computed properties with human-readable names that describe what the conditional is?

Which Computed Properties are you referring to? 
We could further abstract this if statement to be something like `if(isLandusePackage)`